mip/driver_stm32.c

```diff
@@ -63,7 +49,62 @@ static void eth_write_phy(uint8_t addr, uint8_t reg, uint32_t val) {
   while (ETH->MACMIIAR & BIT(0)) spin(1);
 }
 
+static uint32_t get_hclk(void) {
+  struct rcc {
+    volatile uint32_t CR, PLLCFGR, CFGR;
+  } *RCC = (struct rcc *) 0x40023800;
+  uint32_t clk = 0, hsi = 16000000 /* 16 MHz */, hse = 8000000 /* 8MHz */;
```

Comment:
HSI is a fixed value but not common to all MCUs, and the user can trim it.
HSE  depends on user hardware, 8MHz is the one in these Nucleo boards.
We'll need to change HSI for the H7xx, but get_hclk() will also be different. (and the MAC controller is also different)